vsphere/virtual_machine_config_structure.go

```diff
@@ -508,6 +509,117 @@ func flattenExtraConfig(d *schema.ResourceData, opts []types.BaseOptionValue) er
 	return d.Set("extra_config", ec)
 }
 
+// expandVAppConfig reads in all the vapp key/value pairs and returns
+// the appropriate VmConfigSpec.
+//
+// We track changes to keys to determine if any have been removed from
+// configuration - if they have, we add them with an empty value to ensure
+// they are removed from vAppConfig on the update.
+func expandVAppConfig(d *schema.ResourceData) types.BaseVmConfigSpec {
```

Comment:
`BaseVmConfigSpec` is an interface, but we aren't returning anything else other than `VmConfigSpec` so this is abstracting things in a way that they don't need to be abstracted. Can this be changed to return a `*types.VmConfigSpec`?

Reply:
Done.